### ISSUE TITLE
Pooch Dependency Issue

### ISSUE THREAD

**Issue by ozen** [scikit-image/scikit-image#5146]
The following merged PR added dependency to pooch 1.3.0 in data module:
https://github.com/scikit-image/scikit-image/pull/5105/files

The current implementation checks if pooch is installed by trying to import it:

https://github.com/scikit-image/scikit-image/blob/0862e5630a00c151566260f02d118f4ee46bb351/skimage/data/__init__.py#L114-L120

Then uses it as if it's v1.3.0:

https://github.com/scikit-image/scikit-image/blob/0862e5630a00c151566260f02d118f4ee46bb351/skimage/data/__init__.py#L149-L150

This raises error if an older version of pooch such as 1.2.0 is installed.

The code is always executed since data module is imported in skimage init:

https://github.com/scikit-image/scikit-image/blob/0862e5630a00c151566260f02d118f4ee46bb351/skimage/__init__.py#L135

So you get the error even if you don't use data, hence did not install the scikit-image[data] extra.

**Comment by hmaarrfk** [comment]
Yes. I think this is a bug.

If we want to make pooch a soft dependency, then we should check for the version and not use pooch if it is lower than our minimum required specification.

Otherwise, we should make a hard dependency (which I don't think will happen since it was never the intent)

**Comment by hmaarrfk** [comment]
If you have an interest in making the pull request, please tag me and I will take part in the review process.

**Comment by ozen** [comment]
I would like to send a PR. We can use `pooch.__version__`, but comparing it with what we need is tricky. Do you have a suggestion for this? Another option is putting pooch.create inside a try-catch.

**Comment by hmaarrfk** [comment]
Yes version management is tricky. 

However, I think a good comment in the requirements file to ensure it is kept in sync is likely ok.

We can have a birectional comment in both files. Keeping two hardcoded locations is doable.

**Comment by jni** [comment]
@ozen this should be fixed in the just-released 0.18.1! 🎉 I'm going to close this, but please reopen if the issue is not fixed for you! 😊

**Comment by hmaarrfk** [comment]
Thanks for getting this out so quickly @jni

**Comment by iljab** [comment]
Unfortunately the error still occurs with lower versions of pooth (1.1.0)

`AttributeError: module 'pooch' has no attribute '__version__'`

**Comment by hmaarrfk** [comment]
Thanks for the heads up.

I'm inclined to reopen since it is nearly the same issue and I would like to avoid asking you to open a new issue for us.

However, it does really help to open new issues.

**Comment by hmaarrfk** [comment]
```
conda create --name sk python=3.8 --yes
conda activate sk
python -m pip install scikit-image==0.18.1 pooch==1.1.0
python -c "import skimage"
```

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/mark/miniforge3/envs/sk/lib/python3.8/site-packages/skimage/__init__.py", line 135, in <module>
    from .data import data_dir
  File "/home/mark/miniforge3/envs/sk/lib/python3.8/site-packages/skimage/data/__init__.py", line 166, in <module>
    image_fetcher, data_dir = create_image_fetcher()
  File "/home/mark/miniforge3/envs/sk/lib/python3.8/site-packages/skimage/data/__init__.py", line 117, in create_image_fetcher
    pooch_version = pooch.__version__.lstrip('v')
AttributeError: module 'pooch' has no attribute '__version__'

```

**Comment by jni** [comment]
Ouch. Thanks for noticing @iljab!

**Comment by hmaarrfk** [comment]
I believe this issue has been addressed in 0.19.0. Looking at the source, it seems that it uses `hasattr` to check for `__version__` before addressing it.

I believe it should be good to use now.

**Comment by joeranbosma** [comment]
It appears this issue has reappeared. I got the same error for 

```bash
conda create --name sk python=3.8 --yes
conda activate sk
python -m pip install scikit-image==0.21.0 pooch==1.2.0
python -c "import skimage"
```

Errors out with:
```bash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/joeranbosma/miniforge3/envs/sk/lib/python3.8/site-packages/skimage/__init__.py", line 141, in <module>
    from .data import data_dir
  File "<frozen importlib._bootstrap>", line 1039, in _handle_fromlist
  File "/Users/joeranbosma/miniforge3/envs/sk/lib/python3.8/site-packages/lazy_loader/__init__.py", line 82, in __getattr__
    submod = importlib.import_module(submod_path)
  File "/Users/joeranbosma/miniforge3/envs/sk/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/joeranbosma/miniforge3/envs/sk/lib/python3.8/site-packages/skimage/data/_fetchers.py", line 127, in <module>
    _image_fetcher, data_dir = _create_image_fetcher()
  File "/Users/joeranbosma/miniforge3/envs/sk/lib/python3.8/site-packages/skimage/data/_fetchers.py", line 103, in _create_image_fetcher
    image_fetcher = pooch.create(
TypeError: create() got an unexpected keyword argument 'retry_if_failed'
```
